tests/controllers/UsersControllerTest.php

```diff
@@ -226,4 +226,192 @@ public function testThatUserDataIsNotDoubleEscapedOnUserEdit()
+
+        $userMapper = $this->getMockBuilder('\UserMapper')
+            ->setConstructorArgs(array($db,$request))
+            ->getMock();
+
+        $userMapper
+            ->expects($this->once())
+            ->method('isSiteAdmin')
+            ->will($this->returnValue(true));
+
+        $usersController->setUserMapper($userMapper);
+        $usersController->setTrusted($request, $db);
+
+    }
+
+    /**
+     * Ensures that if the setTrusted method is called by an admin,
+     * but the update fails, an exception is thrown
+     *
+     * @return void
+     *
+     * @test
+     * @expectedException        \Exception
+     * @expectedExceptionMessage Unable to update status
+     */
+    public function testSetTrustedWithFailureThrowsException()
+    {
+        $request = $this->getMockBuilder('\Request')->disableOriginalConstructor()->getMock();
+        $request->method('getUserId')->willReturn(2);
+        $request->method('getParameter')
+            ->with("trusted")
+            ->willReturn(true);
+
+        $usersController = new \UsersController();
+        // Please see below for explanation of why we're mocking a "mock" PDO
+        // class
+        $db = $this->getMockBuilder('\JoindinTest\Inc\mockPDO')->getMock();
+
+        $userMapper = $this->getMockBuilder('\UserMapper')
+            ->setConstructorArgs(array($db,$request))
+            ->getMock();
+
+        $userMapper
+            ->expects($this->once())
+            ->method('isSiteAdmin')
+            ->will($this->returnValue(true));
+
+        $userMapper
+            ->expects($this->once())
+            ->method("setTrustedStatus")
+            ->willReturn(false);
+
+        $usersController->setUserMapper($userMapper);
+        $usersController->setTrusted($request, $db);
+
+    }
+
+
+    /**
+     * Ensures that if the setTrusted method is called by an admin,
+     * and the update succeeds, a view is created and null is returned
+     *
+     * @return void
+     */
+    public function testSetTrustedWithSuccessCreatesView()
+    {
+        $request = $this->getMockBuilder('\Request')->disableOriginalConstructor()->getMock();
+        $request->method('getUserId')->willReturn(2);
+        $request->method('getParameter')
+            ->with("trusted")
+            ->willReturn(true);
+
+        $view = $this->getMockBuilder(\JsonView::class)->getMock();
+        $view->expects($this->once())
+            ->method("setHeader")
+            ->willReturn(true);
+
+        $view->expects($this->once())
+            ->method("setResponseCode")
+            ->with(204)
+            ->willReturn(true);
+
+        $request->expects($this->once())
+            ->method("getView")
+            ->willReturn($view);
+
+        $usersController = new \UsersController();
+        // Please see below for explanation of why we're mocking a "mock" PDO
+        // class
+        $db = $this->getMockBuilder('\JoindinTest\Inc\mockPDO')->getMock();
```

Comment:
Presumably because of http://stackoverflow.com/questions/3138946/mocking-the-pdo-object-using-phpunit though i agree we should have that and a set request and DB up in the set up. As it's used everywhere.